opencog/atoms/core/Variables.h

```diff
@@ -132,6 +132,11 @@ struct Variables : public FreeVariables
 	VariableDeepTypeMap _deep_typemap;
 	VariableDeepTypeMap _fuzzy_typemap;
 
+	/// Return true iff all variables are well typed. For now only
+	/// simple types are supported, specifically if some variable is
+	/// simple typed NOTYPE, then it returns false.
```

Comment:
OK, Can you add a comment to the code that the definition of a "well-typed variable" is a variable that has an associated type? Otherwise, I am left guessing as to what "well-typed" really means, in this context.